src/main/java/org/tikv/common/util/ChannelFactory.java

```diff
@@ -188,22 +199,39 @@ public ChannelFactory(
       int keepaliveTime,
       int keepaliveTimeout,
       int idleTimeout,
+      long connRecycleTime,
+      long certReloadInterval,
       String trustCertCollectionFilePath,
       String keyCertChainFilePath,
       String keyFilePath) {
     this.maxFrameSize = maxFrameSize;
     this.keepaliveTime = keepaliveTime;
     this.keepaliveTimeout = keepaliveTimeout;
     this.idleTimeout = idleTimeout;
+    this.connRecycleTime = connRecycleTime;
     this.certContext =
         new OpenSslContext(trustCertCollectionFilePath, keyCertChainFilePath, keyFilePath);
+
+    File trustCert = new File(trustCertCollectionFilePath);
+    File keyCert = new File(keyCertChainFilePath);
+    File key = new File(keyFilePath);
+
+    if (certReloadInterval > 0) {
+      onCertChange();
+      this.certWatcher =
+          new CertWatcher(1, ImmutableList.of(trustCert, keyCert, key), this::onCertChange);
+    } else {
+      this.certWatcher = null;
+    }
   }
 
   public ChannelFactory(
       int maxFrameSize,
       int keepaliveTime,
       int keepaliveTimeout,
       int idleTimeout,
+      long connRecycleTime,
```

Comment:
I don't understand why a connRecycleTime delay is needed when recycling connections. Can you explain this configuration

Reply:
Comments will be added, and the old channel will work for a while even if the certificate is reloaded but it should not be used forever, this recycle time is about to force those expired channels to close and use the latest context after a specific waiting duration.

Reply:
get